src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -132,10 +140,19 @@
     public MonitoredTrip() {
     }
 
-    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) {
+    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISyntaxException {
         queryParams = otpDispatcherResponse.requestUri.getQuery();
         TripPlan plan = otpDispatcherResponse.getResponse().plan;
         itinerary = plan.itineraries.get(0);
+
+        // extract trip time from parsed params
+        List<NameValuePair> parsedParams = getParsedParams();
+        for (NameValuePair parsedParam : parsedParams) {
+            if (parsedParam.getName().equals("time")) {
+                tripTime = parsedParam.getValue();
```

Comment:
At least of these things should also take place with this change:
- Update all `MonitoredTrip` records in Mongo with a value for `tripTime` (and on UI side, make sure that field gets populated too),
- If reading a `MonitoredTrip` record from Mongo with no `tripTime` value set, initialize `tripTime` on first use of `getHour()` or `getMinute()`, which would be more fool-proof.